org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMAttributeFormatter.java

```diff
@@ -43,29 +44,23 @@ public void formatAttribute(DOMAttr attr, int prevOffset, boolean singleAttribut
 			if (isPreserveAttributeLineBreaks() && hasLineBreak(prevOffset, attr.getStart())) {
 				replaceLeftSpacesWithIndentation(indentLevel + 1, prevOffset, attr.getStart(), true, edits);
 				alreadyIndented = true;
+				availableLineWidth = getMaxLineWidth() - getTabSize() * (indentLevel + 1);
 			} else if (isSplitAttributes() && !singleAttribute) {
 				// move the attribute to a new line and indent it.
 				replaceLeftSpacesWithIndentation(indentLevel + getSplitAttributesIndentSize(), prevOffset,
 						attr.getStart(), true, edits);
 				alreadyIndented = true;
+				availableLineWidth = getMaxLineWidth() - getTabSize() * (indentLevel + getSplitAttributesIndentSize());

```

Comment:
As we will have formatterDocument.isMaxLineWidthSupported()

I wonder if we should use it to avoid computing availableLineWidth

If yiu do this check it will be easy to retrieve in the code where available line width is computed

Reply:
In my new changes I have tried to group the availableLineWidth computation together as much as possible but there are still a few that are missing that relies on more logic. I'm not sure if it is productive to do the `isMaxLineWidthSupported()` check for each time I modify availableLineWidth.

Reply:
I like your change, but to be consistent I think you should do that. It will help to retrieve where max lien width is computed by finding references from formatterDocument.isMaxLineWidthSupported() with an IDE